aya/src/maps/mod.rs

```diff
@@ -128,6 +131,7 @@ pub enum MapError {
 pub struct Map {
     pub(crate) obj: obj::Map,
     pub(crate) fd: Option<RawFd>,
+    pub pinned: bool,
```

Comment:
We don't seem to be reading this from anywhere, and we only set it to true when `bpf_object_get`-ting a map, but not after pinning it with `pin()`. Unless I'm missing something, I'd say let's remove this and we can add it back when and if we need it?

Reply:
good catch, i actually meant to include it in the `map.pin` logic as a guard against calling `pin` if we've already called it once before. I'd think this could happen if you wanted to share the same map between two programs that shared the same loader.
i've updated the pin logic to make use of it...